### ISSUE TITLE
Place filter does not seem to be using place poligons

### ISSUE THREAD

**Issue by pandabear-de** [gboeing/osmnx#275]
#### Problem description 
It seems the the function pois_from_place is only filtering on a box instead of the area of the place.

#### What operating system, architecture, Python version, and OSMnx version are you using?
OS10.4, Jupyter Notebook, Python 3.6.5 :: Anaconda, Inc., OSMnx 0.9

#### Complete list of your environment's packages and their versions (for example, run `conda list` or `pip list` then paste the output below, between the two "details" tags)
happy do deliver, if really relevant

# Paste the output of your Python packages and their versions here, between these two "details" tags

</details>

#### Code snippet that independently and completely reproduces the issue

_Overpass_
```Overpass
[out:json][timeout:25];
// gather results
(
  area[name="Karben"];
  node(area)["highway"="bus_stop"]({{bbox}});
);
// print results
out body;
>;
out skel qt;
```
==> 67 POIs

_Python_
```Python
place_name = "Karben"
tags = {
    'highway':'bus_stop',
}
bus_pois = ox.pois_from_place(place=place_name, tags=tags)
cols = ['highway', 'operator', 'network', 'name', 'ref_name', 'bus_lines', 'public_transport', 'shelter', 'covered', 'kerb', 
        'tactile_paving', 'bench', 'wheelchair',  'fixme', 'geometry'
       ]
bus_pois[cols].sort_values('ref_name')
```
==> 110 POIs

Graphical explaination

Here you see where the difference comes from:
```Python
ax = area.plot(facecolor='grey')
bus_pois.plot(ax=ax, color='green', alpha=0.7, markersize=10)
edges.plot(ax=ax, linewidth=0.1, edgecolor='#BC8F8F')
plt.tight_layout()
```

**My Expectation:**
When using pois_from_place I expected the place area to be used for selection as it is done in Overpass. (result 67 pois and not 110)

Are my expectations wrong?

**Comment by gboeing** [comment]
You are correct. The `pois.py` module's `pois_from_place` functionality works inconsistently compared to the rest of OSMnx (e.g., `graph_from_place` or `footprints_from_place`) in that it currently doesn't conform the query area to the shape of the place polygon, but rather queries the polygon's bounding box.

Here is the relevant line: https://github.com/gboeing/osmnx/blob/99cdbe18161b5ce61f42e835bb9a60bff7efc172/osmnx/pois.py#L88

And for comparison, this is how `footprints.py` handles the place query instead: https://github.com/gboeing/osmnx/blob/99cdbe18161b5ce61f42e835bb9a60bff7efc172/osmnx/footprints.py#L117

The pois module should probably be updated to either 1) document this inconsistency in its docstring (temporary fix) or 2) update how it queries by polygon to make it consistent with the footprints module (superior fix).

PRs welcome!

Lastly, CCing @HTenkanen in case he has any input, as he did a lot of work developing this module originally.

**Comment by gboeing** [comment]
> The pois module should probably be updated to either 1) document this inconsistency in its docstring (temporary fix) or 2) update how it queries by polygon to make it consistent with the footprints module (superior fix).

Resolved using option 1 in 5f8a813 but open to implementing option 2 eventually if someone wants to open a PR.

**Comment by kylebarron** [comment]
It seems like it would be simple to first request by bounding box but then filter the responses by the provided polygon? That's not a big change from the current code, but it would clear out any false positives that are returned within the bounding box but outside the polygon that might surprise the user.

**Comment by gboeing** [comment]
fixed in #352
